sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpan.java

```diff
@@ -449,8 +454,62 @@ public ReadWriteSpan recordException(Throwable exception, Attributes additionalA
       additionalAttributes = Attributes.empty();
     }
 
-    addTimedEvent(
-        ExceptionEventData.create(spanLimits, clock.now(), exception, additionalAttributes));
+    AttributeKey<String> exceptionTypeAttributeKey = AttributeKey.stringKey("exception.type");
+    AttributeKey<String> exceptionMessageAttributeKey = AttributeKey.stringKey("exception.message");
+    AttributeKey<String> exceptionStackTraceAttributeKey =
+        AttributeKey.stringKey("exception.stacktrace");
```

Comment:
assuming this is all correct, these keys should be statically allocated as private static final fields to avoid repeated allocations of identical instances.

Reply:
Done!! pushed a new change to replace this.